### PR TITLE
[2/2] Reduce php-fpm image size

### PR DESCRIPTION
**Pull request by ihor-sviziev**
https://github.com/drpayyne/docker-php/pull/3
https://github.com/drpayyne/warden-multi-arch/pull/21

### CODE REVIEW

**Comment by ihor-sviziev** [comment]
@drpayyne, could you please review?

File: images/php-fpm/Dockerfile [hunk 1]
```diff
@@ -10,12 +10,12 @@ ENV COMPOSER_HOME=
 ENV MAILHOG_HOST    mailhog
 ENV MAILHOG_PORT    1025
 
-RUN apt update
-
-RUN apt install -y pv sudo bind9-utils python3-pip bash-completion rsync socat oathtool \
-        wget ack automake autoconf nano vim cron mariadb-server
-
-RUN apt install -y python2 && ln -s /usr/bin/python2 /usr/bin/python
+RUN set -eux; \
```
Comment (drpayyne) [inline images/php-fpm/Dockerfile:13]: What exactly does setting this flag do? I've seen its usage before, but unclear on what exactly it does.

File: images/php-fpm/Dockerfile [hunk 2]
```diff
@@ -10,12 +10,12 @@ ENV COMPOSER_HOME=
...
+    apt-get update -y; \
+    apt-get install -y --no-install-recommends pv sudo bind9-utils python3-pip bash-completion rsync socat oathtool \
+        wget ack automake autoconf nano vim cron mariadb-client python2; \
+    ln -s /usr/bin/python2 /usr/bin/python; \
```
Comment (drpayyne) [inline images/php-fpm/Dockerfile:17]: Any particular reason to add python2 and make it default here?

Comment (ihor-sviziev) [reply images/php-fpm/Dockerfile:17]: It was there all the time

Comment (drpayyne) [reply images/php-fpm/Dockerfile:17]: I don't have much time to test, so I'm just curious.

File: images/php-fpm/Dockerfile (hunk 1 above)
Comment (ihor-sviziev) [reply images/php-fpm/Dockerfile:13]: https://linuxcommand.org/lc3_man_pages/seth.html
>       -e  Exit immediately if a command exits with a non-zero status.
>       -u  Treat unset variables as an error when substituting.
>       -x  Print commands and their arguments as they are executed.

File: images/php-fpm/Dockerfile (hunk 2 above)
Comment (ihor-sviziev) [reply images/php-fpm/Dockerfile:17]: It was just separated to 2 separate RUN commands, I've just merged them together to get smaller image size

File: images/php-fpm/Dockerfile (hunk 1 above)
Comment (drpayyne) [reply images/php-fpm/Dockerfile:13]: Ah, thank you!

File: images/php-fpm/Dockerfile (hunk 2 above)
Comment (drpayyne) [reply images/php-fpm/Dockerfile:17]: Apologies, reviewing on mobile, so I missed the context.

**Comment by drpayyne** [comment]
Thank you for flagging this! I don't get notifications for new PRs/issues unfortunately. Merged. Will rebuild the images later today.

**Comment by ihor-sviziev** [comment]
@drpayyne I see you updated the basic php-fpm images, but not warden m2 images. Could you build them as well?

**Comment by drpayyne** [comment]
Hi @ihor-sviziev, yes I had it in my queue for today and they've now been published. Note - PHP 7.2 image is ready in the docker-php repo but not in Warden Multi Arch. There's an error happening, but I'm unable to look into it at the moment. So we have 7.3, 7.4, and 8.1 updated for Warden.

**Comment by ihor-sviziev** [comment]
Thank you!

**Pull request merged**